src/ui/Combobox/ComboboxValues.ts

```diff
@@ -108,12 +108,16 @@ export class ComboboxValues {
   }
 
   public clearValues() {
+    const shouldEmptyElement = this.hasValues();
     this.mouseIsOverValue = false;
     this.resetKeyboardActiveValue();
-    $$(this.element).empty();
```

Comment:
Why not simply modify this line to only execute if there are values?

```
this.hasValues() && $$(this.element).empty();
```

Or should we check it at the top? Seems strange to do several operations but to exclude this one. e.g.

```

public clearValues() {
    if (!this.hasValues()) {
        return;
    }
     ...
}
```

Reply:
The reason I was keeping so many operations was because unit tests were verifying that analytics would be sent even when there was no values, I wanted to avoid regression. I was told in today's scrum that instead I should fix `Dom`'s `empty`, so I'll do that instead.